src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Icu.cs

```diff
@@ -364,32 +364,51 @@ private static string ConvertIcuTimeFormatString(ReadOnlySpan<char> icuFormatStr
                 char current = icuFormatString[i];
                 switch (current)
                 {
+                    case '\\':
+                        result[resultPos++] = '\\';
```

Comment:
I am not sure what exactly you are trying to do. If pattern has `\` should be kept as it is inside the string and not escaping it by adding extra one. 

For parsing two adjacent two single vertical quotes, it should be easy to check when we get the single quote, check the next character, if it is another single quote, store the two quotes as `\'`. i.e. `''` will be converted to `\'` either inside or outside other quotes. This should simplify the changes here.

Reply:
```C#
                    case '\'':
                        if (i < icuFormatString.Length - 1 && icuFormatString[i + 1] == '\'')
                        {
                            // Double quote, skip the second one
                            result[resultPos++] = '\\';
                            result[resultPos++] = icuFormatString[i++];
                            continue;
                        }

                        result[resultPos++] = icuFormatString[i++];
                        while (i < icuFormatString.Length)
                        {
                            current = icuFormatString[i];
                            if (current == '\'')
                            {
                                if (i < icuFormatString.Length - 1 && icuFormatString[i + 1] == '\'')
                                {
                                    // Double quote, skip the second one
                                    result[resultPos++] = '\\';
                                    result[resultPos++] = icuFormatString[i++];
                                    i++;
                                    continue;
                                }

                                result[resultPos++] = current;
                                break;
                            }

                            result[resultPos++] = current;
                            i++;
                        }
                        break;
```

Reply:
Backslashes are not used as part of escape sequences in the ICU format, so we need to duplicate them or they will be escaped in the .NET format.

Reply:
ok, thanks.

Is there an example of any ICU pattern that uses backslash? I am curious to see such case.

Reply:
There isn't but in case there will be such cases in the future.

Reply:
> To handle parsing two adjacent single vertical quotes, you can simply check the next character when encountering a single quote. If it is another single quote, store the pair as `\'`. Thus, `''` will be converted to `\'` regardless of whether it appears inside or outside other quotes. This approach should simplify the necessary changes.

Agree. I'll work on that later.